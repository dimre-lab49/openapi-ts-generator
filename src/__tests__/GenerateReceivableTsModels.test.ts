import { existsSync, readdirSync, readFileSync } from 'fs';
import { generateTsModels } from '../index';

beforeAll(async () => {
  await generateTsModels(
    'https://d-invr-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
    './output_inr/',
  );
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_inr');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_inr');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an index.ts barrel file', async done => {
  try {
    const fileExist = await existsSync('./output_inr/index.ts');
    expect(fileExist).toBeTruthy();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('IReceivable should be correct', async done => {
  try {
    const file = await readFileSync('./output_inr/receivable.model.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});
