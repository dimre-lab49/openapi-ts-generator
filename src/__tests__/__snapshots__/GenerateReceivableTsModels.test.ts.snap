// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IReceivable should be correct 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-model-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
 // tslint:disable: array-type
 
import { ILineItem } from './line-item.model';
import { IReceivableContact } from './receivable-contact.model';
import { IReceivableDocument } from './receivable-document.model';

export interface IReceivable {
    accountId?: string;
    accountName: string;
    cancelledOnUtc?: Date;
    comments?: string;
    createdBy?: string;
    createdOnUtc?: Date;
    dueDate?: Date;
    emailInvoice?: boolean;
    id?: string;
    isManuallyEdited?: boolean;
    latestGlFileName?: string;
    latestInvoiceFileName?: string;
    lineItemCount?: number;
    lineItemTotal?: number;
    lineItems?: Array<ILineItem>;
    number?: string;
    paidOnDate?: Date;
    previousLineItemTotal?: number;
    receivableContacts?: Array<IReceivableContact>;
    receivableDocuments?: Array<IReceivableDocument>;
    status?: number;
    tenantId: string;
    totalTaxes?: number;
    updatedBy?: string;
    updatedOnUtc?: Date;
}

export enum ReceivableProperties {
    ACCOUNT_ID = 'accountId',
    ACCOUNT_NAME = 'accountName',
    CANCELLED_ON_UTC = 'cancelledOnUtc',
    COMMENTS = 'comments',
    CREATED_BY = 'createdBy',
    CREATED_ON_UTC = 'createdOnUtc',
    DUE_DATE = 'dueDate',
    EMAIL_INVOICE = 'emailInvoice',
    ID = 'id',
    IS_MANUALLY_EDITED = 'isManuallyEdited',
    LATEST_GL_FILE_NAME = 'latestGlFileName',
    LATEST_INVOICE_FILE_NAME = 'latestInvoiceFileName',
    LINE_ITEM_COUNT = 'lineItemCount',
    LINE_ITEM_TOTAL = 'lineItemTotal',
    LINE_ITEMS = 'lineItems',
    NUMBER = 'number',
    PAID_ON_DATE = 'paidOnDate',
    PREVIOUS_LINE_ITEM_TOTAL = 'previousLineItemTotal',
    RECEIVABLE_CONTACTS = 'receivableContacts',
    RECEIVABLE_DOCUMENTS = 'receivableDocuments',
    STATUS = 'status',
    TENANT_ID = 'tenantId',
    TOTAL_TAXES = 'totalTaxes',
    UPDATED_BY = 'updatedBy',
    UPDATED_ON_UTC = 'updatedOnUtc',
}
"
`;

exports[`ReceivableFormGroupFac should be correct 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormGroup, Validators } from '@angular/forms';
export function ReceivableFormGroupFac(): FormGroup {
  return new FormGroup({
    accountId: new FormControl(''),
    accountName: new FormControl('', Validators.required),
    cancelledOnUtc: new FormControl(''),
    comments: new FormControl(''),
    dueDate: new FormControl(''),
    emailInvoice: new FormControl(''),
    id: new FormControl(''),
    isManuallyEdited: new FormControl(''),
    latestGlFileName: new FormControl(''),
    latestInvoiceFileName: new FormControl(''),
    lineItemCount: new FormControl(''),
    lineItemTotal: new FormControl(''),
    lineItems: new FormControl(''),
    number: new FormControl(''),
    paidOnDate: new FormControl(''),
    previousLineItemTotal: new FormControl(''),
    receivableContacts: new FormControl(''),
    receivableDocuments: new FormControl(''),
    status: new FormControl(''),
    tenantId: new FormControl('', Validators.required),
    totalTaxes: new FormControl(''),
  });
};
"
`;
